test/replicatedmap/ReplicatedMapProxyTest.js

```diff
@@ -0,0 +1,130 @@
+var expect = require('chai').expect;
+var HazelcastClient = require('../../lib/index.js').Client;
+var Controller = require('./../RC');
+var Util = require('./../Util');
+var fs = require('fs');
+
+var Promise = require('bluebird');
+
+describe('ReplicatedMap Proxy', function () {
+
+    var cluster;
+    var client;
+    var rm;
+    var ONE_HOUR = 3600000;
+
+    before(function () {
+        this.timeout(10000);
+        var config = fs.readFileSync(__dirname + '/hazelcast_replicatedmap.xml', 'utf8');
+        return Controller.createCluster(null, config).then(function (response) {
+            cluster = response;
+            return Controller.startMember(cluster.id);
+        }).then(function () {
+            return HazelcastClient.newHazelcastClient().then(function (hazelcastClient) {
+                client = hazelcastClient;
+            });
+        });
+    });
+
+    beforeEach(function () {
+        rm = client.getReplicatedMap('test')
+    });
+
+    afterEach(function () {
+        return rm.destroy();
+    });
+
+    after(function () {
+        client.shutdown();
+        return Controller.shutdownCluster(cluster.id);
+    });
+
+    it('puts one entry and gets one entry', function () {
+        return rm.put('key', 'value', ONE_HOUR)
+            .then(function () {
+                return rm.get('key').then(function (val) {
+                    expect(val).to.equal('value');
+                });
+            });
+    });
+
+    it('should contain the key', function () {
+        return rm.put('key', 'value', ONE_HOUR)
+            .then(function (val) {
+                return rm.containsKey('key')
+                    .then(function (res) {
+                        expect(res).to.equal(true);
+                    });
+            });
+    });
+
+    // TODO(zemd): need to be verified separately due to strange NullPointerException from the server
```

Comment:
The problem is exception from server: https://gist.github.com/zemd/e1476c618caf92563dc018597b208682

As I noticed in hazelcast java core there are no so many things for ContainsKeyOperation which can throw such exception, but I haven't figured out.